packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -83,6 +83,7 @@ export const useGridFilter = (
 
       const filteringResult = apiRef.current.unstable_applyStrategyProcessor('filtering', {
         isRowMatchingFilters,
+        linkOperator: filterModel.linkOperator ?? getDefaultGridFilterModel().linkOperator,
```

Comment:
In another PR I would be in favor of creating two interfaces `GridInputFilterModel` and `GridFilterModel`.
One for the model provided by the user.
One for the model in state with the default values (including the default in the items).
To avoid having this kind of fallback deep in the code that we can forget.

Reply:
Good idea, created issue #5589 for that